packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -895,10 +904,30 @@ export abstract class FluidDataStoreContext
 	 * other clients that are up-to-date till seq# 100 may not have them yet.
 	 */
 	protected identifyLocalChangeInSummarizer(eventName: string, type?: string) {
-		if (
-			this.clientDetails.type !== summarizerClientType ||
-			this.localChangesTelemetryCount <= 0
-		) {
+		if (this.clientDetails.type !== summarizerClientType) {
+			return;
+		}
+
+		if (this.abortSummarizerIfLocalChanges) {
+			const error = DataProcessingError.create(
+				"Summarizer with local changes - DataStoreContext",
+				"unexpectedActionReceived",
```

Comment:
I'd suggest something more specific like `identifyLocalChangeInSummarizer` (although it's less interesting for `.create` as opposed to `.wrapIfUnrecognized`)